lib/crypto.js

```diff
@@ -276,13 +276,18 @@ Sign.prototype._write = function _write(chunk, encoding, callback) {
 
 Sign.prototype.update = Hash.prototype.update;
 
-Sign.prototype.sign = function sign(options, encoding) {
-  if (!options)
+Sign.prototype.sign = function sign(privateKey, encoding, options) {
+  if (!privateKey)
     throw new Error('No key provided to sign');
 
-  var key = options.key || options;
-  var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase);
+  if (typeof encoding == 'object' && !options) {
+    options = encoding;
+    encoding = null;
+  }
+
+  var key = privateKey.key || privateKey;
+  var passphrase = privateKey.passphrase || null;
```

Comment:
what about when privateKey is passed in `options`? It would not make sense that `.passPhrase` is supported when the options object is first argument, but not supported when it is second or third argument